Bug: Build fails with avx512 VNNI extensions enabled on windows (solution attached)

### What happened?

I'm trying to build the latest repo on windows with VNNI extensions enabled (for a cpu which supports these extensions):

The `-DGGML_AVX512_VNNI=ON` setting breaks the build.

Windows batch used to build it:

```batch
call "c:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvars64.bat"
git clone https://github.com/ikawrakow/ik_llama.cpp

... varibles omitted

cmake -S . -B %BUILD_DIR% -G "Ninja Multi-Config" ^
  -DGGML_NATIVE=OFF ^
  -DGGML_OPENMP=ON ^
  -DGGML_AVX=OFF ^
  -DGGML_AVX2=ON ^
  -DGGML_AVX512=ON ^
  -DGGML_AVX512_VNNI=ON^
  -DGGML_AVX512_VBMI=ON ^
  -DGGML_AVX512_BF16=ON ^
  -DGGML_CUDA=ON ^
  -DGGML_SCHED_MAX_COPIES=1 ^
  -DGGML_MAX_CONTEXTS=2048 ^
  -DGGML_CUDA_IQK_FORCE_BF16=1 ^
  -DLLAMA_CURL=OFF ^
  -DGGML_CUDA_USE_GRAPHS=ON ^
  -DGGML_CUBLAS=ON ^
  -DCMAKE_CUDA_ARCHITECTURES=86 ^
  -DCUDAToolkit_ROOT="%CUDA_PATH%"

cmake --build %BUILD_DIR% --config Release -j %NINJA_JOBS% --target llama-server
```

And the build fails with the error attached below in a "Relevant log output" section.

I've asked ChatGPT, it proposed a solution which actually fixes the error and the build works.
I've added it to the top of `ggml\src\iqk\iqk_gemm_legacy_quants.cpp` and it solves it.

```cpp
#ifdef _MSC_VER
#include <immintrin.h>

static inline __m512i operator|(__m512i a, __m512i b) { return _mm512_or_si512(a, b); }
static inline __m512i operator&(__m512i a, __m512i b) { return _mm512_and_si512(a, b); }
static inline __m512i operator^(__m512i a, __m512i b) { return _mm512_xor_si512(a, b); }
#endif

//
// ============================== Legacy quants
//
```

Result: build works, llama-server works, inference runs (tested on GLM-4.6 UD-Q4_K_XL).


### Name and Version

llama-server.exe --version
version: 3983 (8a8de91a)
built with MSVC 19.44.35219.0 for


### What operating system are you seeing the problem on?

Windows

### Relevant log output

```shell
C:\PROGRA~1\MICROS~2\2022\COMMUN~1\VC\Tools\MSVC\1444~1.352\bin\Hostx64\x64\cl.exe /nologo /TP -DGGML_BUILD -DGGML_CUDA_DMMV_X=32 -DGGML_CUDA_FUSION=1 -DGGML_CUDA_MIN_BATCH_OFFLOAD=32 -DGGML_CUDA_MMV_Y=1 -DGGML_CUDA_PEER_MAX_BATCH_SIZE=128 -DGGML_CUDA_USE_GRAPHS -DGGML_IQK_FLASH_ATTENTION -DGGML_MAX_CONTEXTS=2048 -DGGML_SCHED_MAX_COPIES=1 -DGGML_SHARED -DGGML_USE_CUDA -DGGML_USE_IQK_MULMAT -DGGML_USE_LLAMAFILE -DGGML_USE_OPENMP -DK_QUANTS_PER_ITERATION=2 -DNDEBUG -D_CRT_SECURE_NO_WARNINGS -D_XOPEN_SOURCE=600 -D__AVX512BF16__ -D__AVX512VBMI__ -D__AVX512VNNI__ -Dggml_EXPORTS -DCMAKE_INTDIR=\"Release\" -ID:\LOG\ik_llama.cpp\ggml\src\..\include -ID:\LOG\ik_llama.cpp\ggml\src\. -external:I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\include" -external:W0 /DWIN32 /D_WINDOWS /W3 /GR /EHsc /MD /O2 /Ob2 /DNDEBUG -std:c++17 /utf-8 /bigobj /arch:AVX512 -openmp /showIncludes /Foggml\src\CMakeFiles\ggml.dir\Release\iqk\iqk_gemm_legacy_quants.cpp.obj /Fdggml\src\CMakeFiles\ggml.dir\Release\ /FS -c D:\LOG\ik_llama.cpp\ggml\src\iqk\iqk_gemm_legacy_quants.cpp 
D:\LOG\ik_llama.cpp\ggml\src\iqk\iqk_gemm_legacy_quants.cpp(1462): error C2676: binary '|': '__m512i' does not define this operator or a conversion to a type acceptable to the predefined operator 
D:\LOG\ik_llama.cpp\ggml\src\iqk\iqk_gemm_legacy_quants.cpp(1462): note: the template instantiation context (the oldest one first) is 
D:\LOG\ik_llama.cpp\ggml\src\iqk\iqk_gemm_legacy_quants.cpp(2007): note: see reference to function template instantiation 'void anonymous-namespace'::mul_mat_q6_0_r4_q8_2<2>(int,const void *,size_t,const DataInfo &,int)' being compiled 
D:\LOG\ik_llama.cpp\ggml\src\iqk\iqk_gemm_legacy_quants.cpp(1463): error C2676: binary '|': '__m512i' does not define this operator or a conversion to a type acceptable to the predefined operator 
D:\LOG\ik_llama.cpp\ggml\src\iqk\iqk_gemm_legacy_quants.cpp(1464): error C2676: binary '|': '__m512i' does not define this operator or a conversion to a type acceptable to the predefined operator 
D:\LOG\ik_llama.cpp\ggml\src\iqk\iqk_gemm_legacy_quants.cpp(1465): error C2676: binary '|': '__m512i' does not define this operator or a conversion to a type acceptable to the predefined operator
```

--- a/ggml/src/iqk/iqk_gemm_legacy_quants.cpp
+++ b/ggml/src/iqk/iqk_gemm_legacy_quants.cpp
@@ -1,9 +1,11 @@
 #include "iqk_gemm_legacy_quants.h"
+
 #include <type_traits>
 
 #ifdef IQK_IMPLEMENT
 
 #include "ggml-impl.h"
+#include "iqk_utils.h"
 
 #define GGML_COMMON_IMPL_C
 #include "ggml-common.h"

--- a/ggml/src/iqk/iqk_utils.h
+++ b/ggml/src/iqk/iqk_utils.h
@@ -80,6 +80,14 @@ static inline float32x4_t v_gelu(float32x4_t x, float32x4_t c1, float32x4_t c2) 
 
 #endif // __ARN_NEON
 
+#if defined(__AVX512F__) && defined(_MSC_VER)
+#include <immintrin.h>
+
+static inline __m512i operator|(__m512i a, __m512i b) { return _mm512_or_si512(a, b); }
+static inline __m512i operator&(__m512i a, __m512i b) { return _mm512_and_si512(a, b); }
+static inline __m512i operator^(__m512i a, __m512i b) { return _mm512_xor_si512(a, b); }
+#endif
+
 #if defined(__AVX512F__) && defined(__AVX512DQ__)
 
 // copy-pasted from Justine Tunney's contribution to llama.cpp

